actionpack/lib/action_dispatch/http/request.rb

```diff
@@ -447,6 +441,10 @@ def check_method(name)
         HTTP_METHOD_LOOKUP[name] || raise(ActionController::UnknownHttpMethod, "#{name}, accepted HTTP methods are #{HTTP_METHODS[0...-1].join(', ')}, and #{HTTP_METHODS[-1]}")
         name
       end
+
+      def default_session
+        Session.disabled(self)
```

Comment:
Setting this `default_session` as disabled without checking the configuration causes any requests built without a session to have this by default and can be surprising. 

If my application has a `config.session_store` set as `:cookie_store` and I create an `ActionDispatch::Request` object its default session will be disabled. Would it be worth adding this information to the changelog as it is likely to break applications using `ActionDispatch::Request` and `ActionDispatch::Request` manually. 

/cc @byroot

Reply:
We happened to run into this in our test suite from both `ActionDispatch::TestRequest` and when using `ActionController::Renderer` from a job. I do like the new behaviour, it could catch bugs where one tries to set a session which has no effect, but I think we should call out this change more explicitly as others are likely to run into it.

Reply:
> any requests built without a session to have this by default and can be surprising.

But that's the idea no. Previously if you were to write into that request's session it would be silently not be persisted, right?

Also I'm curious to hear more about the use case of using `ActionDispatch::Request` manually.

> Would it be worth adding this information to the changelog

I'm totally open to improve the changelog, if you have specific suggestions of things to call out, please let me know.

> We happened to run into this in our test suite from both ActionDispatch::TestRequest

Hum, `TestRequest` should probably assume a working session no? That might be an oversight.